igcollect/rsyslog_impstats.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+"""igcollect - Rsyslog impstats parser
+
+Copyright (c) 2019 InnoGames GmbH
+"""
+
+import re
+from argparse import ArgumentParser
+from time import time
+
+def parse_log():
+    actions = {}
+    args = parse_args()
+
+    with open(args.filename, 'r') as pstats_file:
+        for line in pstats_file:
+            lastmatch = re.findall(r'(action-[0-9]+)(.*discarded\.full.*)', line)
+            if lastmatch:
+                actions [lastmatch[0][0]] = lastmatch[0][1]
```

Comment:
I missing a hint what lastmatch[0][0] and lastmatch[0][1] contains. Easy fix would be to add an example as a comment.
Maybe it is easier to understand if you create two functions like `parse_action_name(lastmatch[0][0])` and `parse_action_stats(lastmatch[0][1])` the late could return a dict containing the `stats_keyvalue_*`

Reply:
The re.findall returns a list of tuples (since we have 2 regex groups (action-[0-9]+) and (.*discarded\.full.*) ), ie. :

[('action-3', '-builtin:omfwd queue: origin=core.queue size=0 enqueued=176432440 full=0 discarded.full=0 discarded.nf=0 maxqsize=92057 ')]

So we're just accessing these elements in the "lastmatch" list and adding these values to a dictionary (and later parsing them) where the key is the action name, and value is the rest of the string which we will later parse to extract the stats/fields we need from it.

Reply:
I've assigned those values to 2 new variables "action_name" and "action_stats" in commit https://github.com/innogames/igcollect/pull/128/commits/f6e2541aba498c489c3c013379edf6b909546b03 , which will hopefully be clearer on what those actually are.